### DISCUSSION TITLE
Can i specify which binary target to include in node_modules?

### DISCUSSION THREAD

**Discussion by ADrejta** [prisma/prisma#6622] [Q&A] [answered]
Hi everyone.

I want to specify a binary target when generating the client based on if I'm working locally or I am deploying an application.

I know that i can specify it in the target in the schema.prisma file like so:
```prisma
generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native", "rhel-openssl-1.0.x"]
}
```

But this includes both the engines when i want to deploy an application.

If i use only the native value then i can work on it only locally and it won't work in the production environment, while if i include only the production binary target i can't work on the application locally. But if i include both of them they get packaged into node_modules and their sizes are quite large (40MB for each binary target).

Can i specify when i run ```prisma generate``` which of the binary targets i want to generate? 
Or is it only in the schema.prisma file at the moment?

**Comment by janpio** [comment, marked as answer]
You can for example load the value for `binaryTargets` from an environment variable - and then change the value in your build or deploy script. An alternative is to remove the unwanted binary from the archive you are deploying.

**Reply by ADrejta** [reply]
Loading them from an environment variable seems like a more hassle free approach. Thanks for your help 🙂

**Reply by janpio** [reply]
Yep, all depends on what you want to achieve, where and why. Happy this works for you 👍

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
